src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+    const { type } = action;
+
+    if (type === 'addProperties') {
+      Object.assign(state, action.extraData);
+    }
+
+    if (type === 'removeProperties') {
+      action.keysToRemove.forEach(key => {
+        delete state[key];
+      });
+    }
+
+    if (type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others